### PR TITLE
Added MTE/PAC error message on MTE/PAC runtime faults/traps

### PR DESCRIPTION
**Pull request by fritzrehde**

### CODE REVIEW

**Comment by fritzrehde** [comment]
For now, I have implemented custom error messages for MTE.

File: crates/runtime/src/traphandlers/macos.rs [hunk 1]
```diff
@@ -434,7 +434,7 @@ unsafe extern "C" fn unwind(
         } else {
             None
         };
-        state.set_jit_trap(wasm_pc, wasm_fp, faulting_addr);
+        state.set_jit_trap(wasm_pc, wasm_fp, faulting_addr.into());
```
Comment (fritzrehde) [inline crates/runtime/src/traphandlers/macos.rs:437]: Does this work on MacOS? Just on a Rust-compiling level, not related to the MTE error message functionality.

File: crates/runtime/src/traphandlers/unix.rs [hunk 2]
```diff
@@ -105,7 +105,26 @@ unsafe extern "C" fn trap_handler(
...
             _ => None,
         };
-        info.set_jit_trap(pc, fp, faulting_addr);
+
+        // TODO: ideally, get these code values from libc somehow
+        // Magic numbers taken from: https://source.android.com/docs/security/test/memory-safety/mte-reports
+        // MTE synchronous mode
+        const SEGV_MTESERR: i32 = 9;
+        // MTE asynchronous mode
+        const SEGV_MTEAERR: i32 = 8;
```
Comment (fritzrehde) [inline crates/runtime/src/traphandlers/unix.rs:114]: Not ideal of course. Maybe these values are in https://rust-lang.github.io/libc/aarch64-unknown-linux-gnu/doc/libc/index.html or sth similar.

File: cranelift/codegen/src/ir/trapcode.rs [hunk 3]
```diff
@@ -49,6 +49,9 @@ pub enum TrapCode {
     /// This trap is resumable.
     Interrupt,
 
+    /// An AArch64-, ARMv8-specific Memory Tagging Extension (MTE) runtime trap.
```
Comment (martin-fink) [inline cranelift/codegen/src/ir/trapcode.rs:52]: I would just write AArch64 here, as AArch64 is ARMv8's 64 bit variant.

File: crates/environ/src/trap_encoding.rs [hunk 4]
```diff
@@ -87,6 +87,9 @@ pub enum Trap {
 
     /// Used to indicate that a trap was raised by atomic wait operations on non shared memory.
     AtomicWaitNonSharedMemory,
+
+    /// An AArch64-, ARMv8-specific Memory Tagging Extension (MTE) runtime trap.
```
Comment (martin-fink) [inline crates/environ/src/trap_encoding.rs:91]: Same here

File: crates/runtime/src/traphandlers/macos.rs (hunk 1 above)
Comment (martin-fink) [reply crates/runtime/src/traphandlers/macos.rs:437]: Yes, works

File: crates/runtime/src/traphandlers/unix.rs [hunk 5]
```diff
@@ -105,7 +105,27 @@ unsafe extern "C" fn trap_handler(
             libc::SIGSEGV | libc::SIGBUS => Some((*siginfo).si_addr() as usize),
             _ => None,
         };
-        info.set_jit_trap(pc, fp, faulting_addr);
+
+        // TODO: ideally, get these code values from libc somehow
+
```
Comment (martin-fink) [inline crates/runtime/src/traphandlers/unix.rs:110]: Remove this comment if we're going to leave this

**Review by martin-fink — COMMENTED** [review]
Looks good! just some small comments

**Comment by fritzrehde** [comment]
I implemented the small changes you requested. Given the lack of documentation on the exception signals/trap PAC sends, I would put implementing that on hold for now. If you want, we can merge this already, and open another branch incase we make progress on also getting error messages to work for PAC.

**Pull request merged**